sql/btree_indexes_on_array_columns.sql

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2019-2024. Ivan Vakhrushev and others.
+ * https://github.com/mfvanek/pg-index-health-sql
+ *
+ * Licensed under the Apache License 2.0
+ */
+
+-- Finds B-tree indexes on array columns
+--
+-- GIN-index should be used instead for such columns
+-- Based on query from https://habr.com/ru/articles/800121/
+select
+    i.indrelid::regclass as table_name,  -- Name of the table
+    i.indexrelid::regclass as index_name, -- Name of the index
+    a.amname    -- index type
```

Comment:
I'd prefer to have information about column
```
    col.attname::text as column_name,
    col.attnotnull as column_not_null
```
like here https://github.com/mfvanek/pg-index-health-sql/blob/521f893a7bf025cc1b6e6992d062c418560b316c/sql/columns_with_json_type.sql#L14

Reply:
To be able instantiate Column model in Java https://github.com/mfvanek/pg-index-health/blob/2470e90ec25ba35468b6852c211f46cf83741d59/pg-index-health-model/src/main/java/io/github/mfvanek/pg/model/column/Column.java#L27

Reply:
When thinking about this requested change, i came up with an idea to rewrite the query:

```
select
    i.indrelid::regclass as table_name,  -- Name of the table
    i.indexrelid::regclass as index_name, -- Name of the index
    a.amname, -- index type
    col.attname as column_name, -- column name
    col.attnotnull as column_not_null -- column not null
from pg_catalog.pg_index as i
    inner join pg_catalog.pg_stat_all_indexes psai on i.indexrelid = psai.indexrelid
    inner join pg_catalog.pg_class as ic on i.indexrelid = ic.oid
    inner join pg_catalog.pg_am as a on ic.relam = a.oid and a.amname = 'btree'
    inner join pg_catalog.pg_attribute as col on i.indrelid = col.attrelid and col.attnum = any((string_to_array(i.indkey::text, ' ')::int2[])[:indnkeyatts])
    inner join pg_catalog.pg_type as typ on typ.oid = col.atttypid
where psai.schemaname = :schemaname and typ.typcategory = 'A';
```

What do you think about taking this one? It differs from the [initial query](https://github.com/mfvanek/pg-index-health/issues/364#issue-2229911315) taken from habr article.

Reply:
1. Results should be ordered somehow
2. pg_catalog.pg_namespace is more preferable way to get a schema

 ```sql
select
    i.indrelid::regclass as table_name,  -- Name of the table
    i.indexrelid::regclass as index_name, -- Name of the index
    col.attname as column_name, -- column name
    col.attnotnull as column_not_null -- column not null
from pg_catalog.pg_index as i
         inner join pg_catalog.pg_class as ic on i.indexrelid = ic.oid
         inner join pg_catalog.pg_namespace nsp on nsp.oid = ic.relnamespace
         inner join pg_catalog.pg_am as a on ic.relam = a.oid and a.amname = 'btree'
         inner join pg_catalog.pg_attribute as col on i.indrelid = col.attrelid and col.attnum = any((string_to_array(i.indkey::text, ' ')::int2[])[:indnkeyatts])
         inner join pg_catalog.pg_type as typ on typ.oid = col.atttypid
where
    nsp.nspname = :schema_name_param::text and
    typ.typcategory = 'A'
order by ic.oid::regclass::text, i.indexrelid::regclass::text;
```

Reply:
Thanks a lot! I rewrote the query.